ext/ffi_c/DynamicLibrary.c

```diff
@@ -27,9 +27,14 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+
+// ruby.h first to provide _GNU_SOURCE which gives us dlvsym if available
+#include <ruby.h>
+
 #include <sys/types.h>
 #include <stdio.h>
 #include <stdint.h>
+
 #if (defined(_WIN32) || defined(__WIN32__)) && !defined(__CYGWIN__)
 # include <winsock2.h>
```

Comment:
,m,,,,

Reply:
Not sure what ^^ means but the alternative to this change in include order is to define _GNU_SOURCE specifically and accept that ruby.h generates a warning about it being redefined.